README.md

```diff
@@ -1,240 +1,229 @@
+Pull request must be approved by at least one engineering team members.
+
+# Authoring code
+
+We recommend using [Visual Studio Code](https://code.visualstudio.com/) for authoring.
+
+
+# Building code
+
+## Linux Agent
+
+### Install Pre-requisites
+
+1. Install go1.14.1, build dependencies and docker if you dont have installed already on your dev machine
+```
+bash ~/Docker-Provider/scripts/buildscripts/install-build-pre-requisites.sh
+```
+
+### Build Docker Provider Shell Bundle
+
+1. Set GOPATH Environment variable and get the go dependencies
+```
+export GOPATH=~/Docker-Provider/source/code/go #Set this based on your repo path
+cd ~/Docker-Provider/source/code/go/src/plugins
+go get
+```
+2. Build the code  with below commands
+```
+cd ~/Docker-Provider/build/linux
+bash ./configure --enable-ulinux
+make
+```
+If build successful, you should see docker-cimprov-x.x.x-x.universal.x86_64.sh under ~/Docker-Provider/target/Linux_ULINUX_1.0_x64_64_Release/
+  > Note: x.x.x-x is the version of the docker provider which is determined from version info in version file
+
+### Build Docker Image
+
+1.  Navigate to below directory to build the docker image
+  ```
+  cd ~/Docker-Provider/kubernetes/linux/
+  ```
+2. Upload docker-cimprov-x.x.x-x.universal.x86_64.sh from ~/Docker-Provider/target/Linux_ULINUX_1.0_x64_64_Release/ to azure blob storage account blob
+
+ ```
+ AZURE_SUBSCRIPTIONID=<azure-subscription-id> # subscriptionId for the azure storage account (new or exist)
+ STORAGE_ACCOUNT_RG=<resource-group>
+ STORAGE_ACCOUNT_LOCATION=<location>
+ STORAGE_ACCOUNT_NAME=<account-name>
+ STORAGE_ACCOUNT_BLOB_NAME=<blob-name> # for example agentshellbundle
+
+ # login to azure interactively and set the subscription id
+ az login --use-device-code
+ az account set -s $AZURE_SUBSCRIPTIONID
+
+ # create rg and storage account if this doesnt exist one already
+ az group create -n $STORAGE_ACCOUNT_RG -l $STORAGE_ACCOUNT_LOCATION
+ az storage account create -n $STORAGE_ACCOUNT_NAME -g $STORAGE_ACCOUNT_RG -l $STORAGE_ACCOUNT_LOCATION
+
+ # create blob container
+ az storage container create --account-name $STORAGE_ACCOUNT_NAME --name $STORAGE_ACCOUNT_BLOB_NAME --auth-mode login --public-access container
+
+ # upload docker-cimprov shell bundle to storage account. please specify the correct version
+ az storage blob upload --account-name $STORAGE_ACCOUNT_NAME --container-name $STORAGE_ACCOUNT_BLOB_NAME --name docker-cimprov-10.0.0-0.universal.x86_64.sh --file  ~/Docker-Provider/target/Linux_ULINUX_1.0_x64_64_Release/docker-cimprov-10.0.0-0.universal.x86_64.sh
+ # replace the placeholders in this url
+ https://$STORAGE_ACCOUNT_NAME.blob.core.windows.net/$STORAGE_ACCOUNT_BLOB_NAME/docker-cimprov-10.0.0-0.universal.x86_64.sh
+
+ # verify this url valid and exist by running the curl --head command as below
+ curl --head https://mydevomsagentsa.blob.core.windows.net/agentshellbundle/docker-cimprov-10.0.0-0.universal.x86_64.sh
```

Comment:
> $STORAGE_ACCOUNT_NAME = mydevomsagentsa
> $STORAGE_ACCOUNT_BLOB_NAME = agentshellbundle
> 
> So we're trying to verify if 'curl --head https://$STORAGE_ACCOUNT_NAME.blob.core.windows.net/$STORAGE_ACCOUNT_BLOB_NAME/docker-cimprov-10.0.0-0.universal.x86_64.sh' is valid right?

> yes, since this constructed by handy so --head validate the url correct or not.